src/main/java/io/gravitee/policy/v3/jwt/JWTPolicyV3.java

```diff
@@ -267,6 +292,63 @@ private CompletableFuture<JWTClaimsSet> validate(ExecutionContext executionConte
             );
         }
 
-        return keyProcessor.process(signature, token);
+        CompletableFuture<JWTClaimsSet> process = keyProcessor.process(signature, token);
+        return process.thenApply(jwtClaimsSet -> {
+            // Validate confirmation method
+            JWTPolicyConfiguration.ConfirmationMethodValidation confirmationMethodValidation =
+                configuration.getConfirmationMethodValidation();
+            if (
+                confirmationMethodValidation != null &&
+                confirmationMethodValidation.getCertificateBoundThumbprint().isEnabled() &&
+                !isValidCertificateThumbprint(
+                    jwtClaimsSet,
+                    executionContext.request().sslSession(),
+                    executionContext.request().headers(),
+                    confirmationMethodValidation.isIgnoreMissing(),
+                    confirmationMethodValidation.getCertificateBoundThumbprint()
+                )
+            ) {
+                throw new InvalidCertificateThumbprintException("Confirmation method validation failed");
+            }
+            return jwtClaimsSet;
+        });
+    }
+
+    protected static boolean isValidCertificateThumbprint(
+        final JWTClaimsSet jwtClaimsSet,
+        final SSLSession sslSession,
+        final io.gravitee.gateway.api.http.HttpHeaders headers,
+        final boolean ignoreMissingCnf,
+        final JWTPolicyConfiguration.CertificateBoundThumbprint certificateBoundThumbprint
+    ) {
+        try {
+            String tokenThumbprint = Optional
+                .ofNullable(jwtClaimsSet.getJSONObjectClaim(CLAIMS_CNF))
+                .map(cnf -> cnf.get(CLAIMS_CNF_X5T))
+                .map(Object::toString)
+                .orElse(null);
+
+            // Ignore empty configuration method
+            if (!StringUtils.hasText(tokenThumbprint) && ignoreMissingCnf) {
+                return true;
+            } else if (!StringUtils.hasText(tokenThumbprint) && !ignoreMissingCnf) {
+                return false;
+            }
+
+            // Compute client certificate thumbprint
+            Optional<X509Certificate> clientCertificate;
+            if (certificateBoundThumbprint.isExtractCertificateFromHeader()) {
+                clientCertificate = CertificateUtils.extractCertificate(headers, certificateBoundThumbprint.getHeaderName());
+            } else {
+                clientCertificate = CertificateUtils.extractPeerCertificate(sslSession);
+            }
```

Comment:
**question**: Maybe I'm wrong but as far as I understand, this is not consistent with the behaviour described into the JIRA ticket "`When specified, the JWT policy will try to extract the client certificate using the specified HTTP header. In case of a missing header, the policy will fall back to extracting it from the current SSL session`". Here we only check the config. In addition according to the JIRA interpretation, we have to fallback to the sslSession if the "header name" is null or if the "Header value" is null.

Reply:
The JIRA ticket is not up to date with what we have agreed on.